### PR TITLE
shortcut erase_marked_points when possible

### PR DESCRIPTION
**Pull request by videlec**
When there is no marked point, no need to erase them :)

Also the method only works for translation surfaces. No need to declare it on `SimilaritySurface`.

### CODE REVIEW

**Comment by videlec** [comment]
@saraedum asv seems not happy
```
Failure creating environment for conda-py3.10-cppyy_=2.0.0,_3-cxx-compiler-gap-defaults-pip+flipper-pyeantic1.0.3-pyexactreal2.2.1-pyflatsurf3.9.3-python-sagelib9.4-scipy-surface-dynamics_=0.4.7
```
Seems a bit similar to the [issue we had in admcycles](https://gitlab.com/jo314schmitt/admcycles/-/issues/78).

**Comment by saraedum** [comment]
Seems you figured it out :)

**Comment by videlec** [comment]
> Seems you figured it out :)

Sort of. Is it expected that asv specifies the python version inside the environment it creates for running the benchmark? Even if we set `python: ["3.8"]` in `asv.conf.json` asv will create an environment with python 3.10 (or the version which is used to run asv).

**Comment by videlec** [comment]
The function `erase_marked_points` turns out to take a lot of time (for no reason) when building unfoldings of complicated triangles. See #122 to improve `erase_marked_points` when it is needed.

File: flatsurf/geometry/translation_surface.py [hunk 1]
```diff
@@ -474,6 +474,62 @@ def j_invariant(self):
...
+            sage: from flatsurf import EquiangularPolygons, similarity_surfaces
+            sage: E = EquiangularPolygons((10, 7, 2, 2, 2, 1))
+            sage: P = E((1, 1, 2, 3), normalized=True)
+            sage: B = similarity_surfaces.billiard(P, rational=True)
+            sage: S_mp = B.minimal_cover(cover_type="translation")
+            sage: S = S_mp.erase_marked_points() # long time (3s), optional: pyflatsurf
+
+        """
+        if all(a != 1 for a in self.angles()):
+            # no fake marked points
```
Comment (saraedum) [inline flatsurf/geometry/translation_surface.py:524]: What's the meaning of fake here?

File: flatsurf/geometry/translation_surface.py [hunk 2]
```diff
@@ -638,6 +694,8 @@ def left(self, label):
     def _repr_(self):
         return "Origami defined by r=%s and u=%s"%(self._r,self._u)
 
+
+
```
Comment (saraedum) [inline flatsurf/geometry/translation_surface.py:698]:
```suggestion
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Install dependencies
         shell: bash -l {0}
         run: |
-          mamba install -n test -y pip
+          mamba install -n test -y python=3.9.10 pip
```
Comment (saraedum) [inline .github/workflows/benchmark.yml:23]: I think you should set `python-version: "3.9"` on the setup-miniconda instead.

**Review by saraedum — APPROVED** [review]

File: flatsurf/geometry/translation_surface.py (hunk 2 above)
Comment (videlec) [reply flatsurf/geometry/translation_surface.py]: I thought that Python recommands 2 linebreaks between class definitions. (at least this is what pyflakes suggests by default)

**Pull request merged**